lib/full-site-editing/block-templates.php

```diff
@@ -386,41 +386,43 @@ function gutenberg_get_block_template( $id, $template_type = 'wp_template' ) {
  * @param string $post_type     Post type.
  * @return string The original, desired slug.
  */
-function gutenberg_filter_wp_template_unique_post_slug( $slug, $post_ID, $post_status, $post_type ) {
-	if ( 'wp_template' !== $post_type || 'wp_template_part' !== $post_type ) {
```

Comment:
The code in this function was never firing because of this early return condition ( `||` should be `&&` ).  With `||` this would always return early, since at least one of those conditions will always be true.

Reply:
Nice catch!